### PR TITLE
RNA Seq Counts to Viz in R lesson

### PR DESCRIPTION
**Pull request by fpsom**
This is the lesson produced during the [Gallantries project](https://galaxy-carpentries.github.io/gallantries/). It's based on the Carpentries [Intro to R and RStudio for Genomics](https://datacarpentry.org/genomics-r-intro/) adapted for the Galaxy ref-based RNA Seq lesson.

### CODE REVIEW

**Comment by tobyhodges** [comment]
TODO:

- [ ] describe how to launch RStudio within Galaxy
- [ ] describe how to load data into R from Galaxy history (section "Importing tabular data into R"
- [ ] describe how to write out a file to Galaxy (section(s) "Saving your data frame to a file" and/or "Exporting data" [dplyr])
- [ ] add some Galaxy-specific context to the conclusion
- [ ] describe why it's helpful to be able to access an interactive RStudio session within Galaxy (in `key_points` of YAML front matter)

**Comment by bebatut** [comment]
After discussion with @fpsom, we will probably split this tutorial into 3:
- "Introduction to RStudio IDE" as a new tutorial for UI topic
- "R basics" as a new tutorial in Introduction topic, as it may be useful for other topics than transcriptomics
- Data frame manipulation, dyplr, ggplot staying there as transcriptomic datasets and plot are used

**Comment by bebatut** [comment]
I made the re-arrangement as mentioned before
@fpsom @mkuzak could you check if the changes make sense?

We need the tutorials quite urgently (on Wednesday for a workshop) so it will be great to merge them tomorrow. But we need first: how to launch RStudio, how to get data from and to history, etc.

I "disabled" the tutorials: They will be online but not listed in the topic pages. We can then use/test them during the workshop and fix them right after

**Comment by mkuzak** [comment]
thank you @bebatut , I'll review it now. Am I looking at the right thing? https://github.com/fpsom/training-material/blob/rnaseq-r/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.md

**Comment by bebatut** [comment]
@mkuzak the order will be:

1. [Rstudio in Galax](https://github.com/galaxyproject/training-material/blob/a8f1a439973d3c09c27ed3f8fa85bfeb7292e9a3/topics/galaxy-ui/tutorials/rstudio/tutorial.md)
2. [R basics](https://github.com/galaxyproject/training-material/blob/a8f1a439973d3c09c27ed3f8fa85bfeb7292e9a3/topics/introduction/tutorials/r-basics/tutorial.md)
3. [RNA-seq counts to viz with R](https://github.com/galaxyproject/training-material/blob/a8f1a439973d3c09c27ed3f8fa85bfeb7292e9a3/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.md)

File: topics/galaxy-ui/tutorials/rstudio/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+> >
+> > While your search results may return several tests, we list a few you might
+> > find:
+> > 1. Student-t test: `stats::TDist`
+> > 2. mixed linear model: `stats::lm.glm`
+> >
+> {: .solution}
+{: .question}
+
+We will not discuss now on where to look for the libraries and packages that contain functions you want to use. For now, be aware that two important ones are:
```
Comment (mkuzak) [inline topics/galaxy-ui/tutorials/rstudio/tutorial.md]:
```suggestion
We will not discuss now, where to look for the libraries and packages that contain functions you want to use. For now, be aware that two important ones are:
```

File: topics/introduction/tutorials/r-basics/disclaimer.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+>
+> We believe that every learner can **achieve competency with R**. You have reached competency when you find that you are able to **use R to handle common analysis challenges in a reasonable amount of time** (which includes time needed to look at learning materials, search for answers online, and ask colleagues for help). As you spend more time using R (there is no substitute for regular use and practice) you will find yourself gaining competency and even expertise. The more familiar you get, the more complex the analyses you will be able to carry out, with less frustration, and in less time - the fantastic world of R awaits you!
+>
+> Nobody wants to learn how to use R. People want to learn how to use R to analyze their own research questions! Ok, maybe some folks learn R for R's sake, but these lessons assume that you want to start analyzing genomic data as soon as possible. Given this, there are many valuable pieces of information about R that we simply won't have time to cover. Hopefully, we will clear the hurdle of giving you just enough knowledge to be dangerous, which can be a high bar in R! We suggest you look into the additional learning materials in the box below.
```
Comment (mkuzak) [inline topics/introduction/tutorials/r-basics/disclaimer.md]:
```suggestion
> Nobody wants to learn how to use R. People want to learn how to use R to answer their own research questions! Ok, maybe some folks learn R for R's sake, but these lessons assume that you want to start analyzing genomic data as soon as possible. Given this, there are many valuable pieces of information about R that we simply won't have time to cover. Hopefully, we will clear the hurdle of giving you just enough knowledge to be dangerous, which can be a high bar in R! We suggest you look into the additional learning materials in the box below.
```

File: topics/introduction/tutorials/r-basics/r_advantages.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+At more than 20 years old, R is fairly mature and [growing in popularity](https://www.tiobe.com/tiobe-index/r/). However, programming isn't a popularity contest. Here are key advantages of analyzing data in R:
+
+- **R is [open source](https://en.wikipedia.org/wiki/Open-source_software)**
+
+    This means R is free - an advantage if you are at an institution where you have to pay for your own MATLAB or SAS license. Open source, is important to your colleagues in parts of the world where expensive software in inaccessible. It also means that R is actively developed by a community (see [r-project.org](https://www.r-project.org/)), and there are regular updates.
```
Comment (mkuzak) [inline topics/introduction/tutorials/r-basics/r_advantages.md]: Open Source is not the same as free. Yes R is free, but software does not have to be OS to be free. I think we're mixing up terms here. Advantage of being free is that you don't have to pay for it. Advantage of being Open Source is that you can see and reuse the source.

File: topics/galaxy-ui/tutorials/rstudio/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+  - tobyhodges
+---
+
+
+# Introduction
+{:.no_toc}
+
+> ### {% icon comment %} Comment
+>
+> This tutorial is significantly based on [the Carpentries](https://carpentries.org) ["Intro to R and RStudio for Genomics"](https://carpentrieslab.github.io/genomics-r-intro/) lesson
```
[the remark this answers is not in the document]
Comment (bebatut) [reply topics/galaxy-ui/tutorials/rstudio/tutorial.md]: I update the link to https://datacarpentry.org/genomics-r-intro/01-introduction/index.html

File: topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,1583 @@
...
+{: .hands_on}
+
+`dplyr` also provides useful functions to select columns based on their names. For instance, `starts_with()` allows you to select columns that starts with specific letters. 
+
+> ### {% icon hands_on %} Hands-on: Select columns that end with the letter "P"
+> 
+> 1. Select columns that end with the letter "P"
+>
+>    ```R
+>    > select(annotatedDEgenes, starts_with("P."))
```
[the remark this answers is not in the document]
Comment (bebatut) [reply topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.md]: Nice catch. Starting with P

File: topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,1583 @@
...
+We can do this with **pipes**. Pipes, are a fairly recent addition to R. They let you take the output of one function and send it directly to the next, which is useful when you need to do many things to the same data set. It was possible to do this before pipes were added to R, but it was much messier and more difficult. Pipes in R look like `%>%` and are made available via the `magrittr` package, which is installed as part of `dplyr`. 
+
+> ### {% icon comment %} Pipes on RStudio
+>
+> With RStudio, you can type the pipe with
+> - <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>M</kbd> if you're using a PC
+> - <kbd>Cmd</kbd> + <kbd>Shift</kbd> + <kbd>M</kbd> if you're using a Mac
+{: .comment}
+
+> ### {% icon hands_on %} Hands-on: Select and filter in the same ti,e
```
[the remark this answers is not in the document]
Comment (bebatut) [reply topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.md]: I removed the `in the same time`

File: topics/introduction/tutorials/r-basics/r_advantages.md (hunk 3 above)
Comment (bebatut) [reply topics/introduction/tutorials/r-basics/r_advantages.md]: I agree with you. This was a copy/paste from the Data Carpentry content: https://datacarpentry.org/genomics-r-intro/01-introduction/index.html

I rephrased this part

**Comment by fpsom** [comment]
Yes. That is the current planning. But generally speaking, I do the before-lunch session, you do the after-lunch one. :)

**Comment by bebatut** [comment]
So the inverse @fpsom 1 and 2 (morning), @mkuzak 3 (afternoon)?

**Comment by fpsom** [comment]
Actually no - by accident I mis-typed. @mkuzak is doing morning, @fpsom is doing afternoon. Sorry for the confusion... :/

**Pull request merged**